types/index.d.ts

```diff
@@ -309,7 +309,7 @@ declare module 'mongoose' {
     eachPath(fn: (path: string, type: SchemaType) => void): this;
 
     /** Defines an index (most likely compound) for this schema. */
-    index(fields: IndexDefinition, options?: IndexOptions): this;
+    index(fields: IndexDefinition, options?: Omit<IndexOptions, 'unique'> & { unique?: boolean | undefined | [true, string] }): this;
```

Comment:
```suggestion
    index(fields: IndexDefinition, options?: Omit<IndexOptions, 'unique'> & { unique?: boolean | [true, string] }): this;
```
i think this `undefined` can be omitted because property is already marked `?`

Reply:
Confirmed. I thought the `undefined` was necessary to support `unique: undefined` as opposed to omitting `unique`, but that doesn't look to be the case.